### ISSUE TITLE
[BUG]  SyntaxError: Unexpected token .

### ISSUE THREAD

**Issue by VeaceslavB** [npm/nopt#130]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

get build error when adding ios in cordova

13:30:16  Using /private/var/folders/wv/d7j3q171735794lxc610xqy80000gp/T/tmp.YLxIVwB6/mobile/cordova/node_modules/cordova CLI version (12.0.0) 
13:30:16  Setting Current Working Directory (CWD) to cordova
13:30:16  Running:/private/var/folders/wv/d7j3q171735794lxc610xqy80000gp/T/tmp.YLxIVwB6/mobile/cordova/node_modules/cordova/bin/cordova platform add ios
13:30:16  /private/var/folders/wv/d7j3q171735794lxc610xqy80000gp/T/tmp.YLxIVwB6/mobile/cordova/node_modules/nopt/lib/nopt-lib.js:64
13:30:16    const StringType = typeDefs.String?.type
13:30:16                                       ^
13:30:16  
13:30:16  SyntaxError: Unexpected token .
13:30:16      at Module._compile (internal/modules/cjs/loader.js:760:23)
13:30:16      at Object.Module._extensions..js (internal/modules/cjs/loader.js:827:10)
13:30:16      at Module.load (internal/modules/cjs/loader.js:685:32)
13:30:16      at Function.Module._load (internal/modules/cjs/loader.js:620:12)
13:30:16      at Module.require (internal/modules/cjs/loader.js:723:19)
13:30:16      at require (internal/modules/cjs/helpers.js:14:16)
13:30:16      at Object.<anonymous> (/private/var/folders/wv/d7j3q171735794lxc610xqy80000gp/T/tmp.YLxIVwB6/mobile/cordova/node_modules/nopt/lib/nopt.js:1:13)
13:30:16      at Module._compile (internal/modules/cjs/loader.js:816:30)
13:30:16      at Object.Module._extensions..js (internal/modules/cjs/loader.js:827:10)
13:30:16      at Module.load (internal/modules/cjs/loader.js:685:32)
13:30:16  >> Error
13:30:16  Error-> with Parallel tasksError

### Steps To Reproduce

1. In this environment...
2. With this config...
3. Run '...'
4. See error...

### Environment

- npm: 6.14.18
- Node: 14.21.3
- OS: MacOS Ventura 13.4
- platform: Macbook Pro

**Comment by wraithgar** [comment]
You are using an unsupported version of node

```sh
$ npm view nopt engines
{ node: '^14.17.0 || ^16.13.0 || >=18.0.0' }
```

**Comment by robin-huss** [comment]
This is indeed an issue when using an unsupported node version. I had the exact same issue.

Turns out i was running both nvm-windows and nodist. I configured nvm-windows to use 18.17.1, but nodist was configured with node 11.13.0. Somehow the `node -v` returns 18.17.1, but the actual node version used was 11.13.0. which leads to the mentioned SyntaxError.

I know this issue is closed, but i wanted to post this here to hopefully save others from having to dig into this.
